### ISSUE TITLE
Licensing question on Fake6502 implementation

### ISSUE THREAD

**Issue by gek169** [commanderx16/x16-emulator#464]
[Fake6502.c](https://github.com/commanderx16/x16-emulator/blob/master/src/cpu/fake6502.c) contains a public domain license notice.

However, this license notice is not duplicated in the rest of the files in the directory.

So there is a grey area here: Is this code really public domain?

It *says* it's in the public domain... but only in 1 source file.

I am currently using portions of your repository's code which I am releasing into the public domain, which contains my own fixes, particularly for decimal mode behavior.

https://github.com/C-Chads/MyLittle6502

in the file fake65c02.h.

It is very important to me to be able to release my work into the public domain, and since the original by Mike Chambers was in the public domain, and your code actually keeps the "this source code is in the public domain" notice, I wondered if all of it was in the public domain or not.

I can of course give you my fixes for free to incorporate into your BSD 2 clause repository.

I can re-license under the BSD, but I really do not want to do that.

~~~~~~~~~~~~~~~~~~~
Why public domain?
How about a technical argument...
~~~~~~~~~~~~~~~~~~~
Code with attribution-requirement licenses generally must reproduce the attribution in any derivative work. This adds to the binary size of the distributed work, which produces an unnecessary burden on the end user to keep information on their devices that they will probably never use.

For instance, if a 1 kilobyte game program for the Vic 20 is under the MIT, then it must have the MIT license in the distribute version. This would take up a rather large amount of space for a Vic 20 and might even interfere with usage of the program
~~~~~~~~~~~~~

TL;DR it is unclear if the code in the cpu folder is BSD 2 clause or public domain or what.


~DMHSW

**Comment by gek169** [comment]
Actually, it seems like you've already incorporated changes to your implementation from MY repository! Interesting...
